### PR TITLE
Gulp migration

### PR DESCRIPTION
**Pull request by m-suchorski**
Grunt changed to Gulp.

## Description
App is now built using Gulp and because of that there is no need to install Ruby - we don't need Compass anymore. Refreshing after chaning SCSS/JS files is also much faster.

## Motivation and Context
This change solves the issue with slow refresh/build times and it also gets rid of Compass.
User now can also go directly to localhost:9000 (by default) and he will be redirected to the report view, because I've added an index.html file.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: report/src/main/webapp/assets/sass/main.scss [hunk 1]
```diff
@@ -15,8 +15,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-@import "compass";
-@import "../../node_modules/bootstrap-sass-twbs/assets/stylesheets/_bootstrap.scss";
+//@import "compass";
```
Comment (tkaik) [inline report/src/main/webapp/assets/sass/main.scss]: Please remove this line if it's not needed

File: report/src/main/webapp/package.json [hunk 2]
```diff
@@ -1,29 +1,46 @@
...
-    "grunt-contrib-jshint": "~1.0.0",
-    "grunt-contrib-uglify": "~0.2.4",
-    "grunt-contrib-watch": "^0.6.1",
-    "grunt-includes": "^0.4.4",
-    "grunt-open": "^0.2.3",
-    "lodash": "^4.14.2",
-    "matchdep": "^0.3.0",
-    "time-grunt": "^1.0.0"
+  "name": "aet",
+  "version": "2.1.16",
```
Comment (tkaik) [inline report/src/main/webapp/package.json]: Why `2.1.16`? Latest AET release is `2.1.6` and currently we have `2.1.7-SNAPSHOT` on master branch

File: report/src/main/webapp/index.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<script>
+    (function(){
```
Comment (Skejven) [inline report/src/main/webapp/index.html]: This is not generic behaviour. There should be no such logic in the main `index.html`page.

File: report/src/main/webapp/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,13796 @@
+{
```
Comment (Skejven) [inline report/src/main/webapp/package-lock.json]: This file looks like generated one. Do we need this here?

Comment (m-suchorski) [reply report/src/main/webapp/package-lock.json]: Yes, my bad. Gonna get rid of it.

File: .gitignore [hunk 5]
```diff
@@ -36,6 +36,10 @@ atlassian-ide-plugin.xml
 /report/src/main/webapp/.sass-cache/
 # auto-generated assets
 /report/src/main/webapp/assets/libs
+# auto-generated css file
+/report/src/main/webapp/assets/css
+# auto-generated package-lock file
+/report/src/main/webapp/package-lock.json
```
Comment (wiiitek) [inline .gitignore]: This file should be commited in my opinion.
Frontend maven plugin reports it as:
```
[ERROR] npm notice created a lockfile as package-lock.json. You should commit this file.
```
please consult i.e. @mateuszluczak

Comment (Skejven) [reply .gitignore]: If it should be commited, is it possible to make it smaller?
It had 13 000 lines.

File: report/src/main/webapp/assets/sass/_typography.scss [hunk 6]
```diff
@@ -144,8 +144,8 @@
   font-style: italic;
 }
 
-input[type=text] {
-  @include input-placeholder {
-    color: #85898e;
-  }
-}
\ No newline at end of file
+// input[type=text] {
```
[the remark this answers is not in the document]
Comment (m-suchorski) [reply report/src/main/webapp/assets/sass/_typography.scss]: Done

File: .gitignore (hunk 5 above)
Comment (adamkosmala) [reply .gitignore]: it should be commited, that's true, but I've never seen anywhere minified version of this file- I'd say, it won't hurt to put it as it is

**Pull request merged**